docs/dev/atomics.md

```diff
@@ -0,0 +1,52 @@
+# Atomic Access
+
+tl;dr: use `github.com/uber-go/atomic` for all atomic access.  Use pointers (`*atomic.Uint64`, etc.) in structs to ensure proper alignment.
+
+## Prefer Not
+
+First, atomics are a _very_ low-level concept and full of subtle gotchas and dramatic performance differences across platforms.
+If you can use something higher-level, such as something from the `sync` package, prefer to do so.
+Otherwise, if you are in search of performance, be sure to benchmark your work carefully and on multiple platforms.
+
+It's tempting to use an atomic to avoid complaints from the race detector, but this is almost always a mistake -- it is merely hiding your race from the race detector.
+Consider carefully why the implementation is racing, and try to address that behavior directly.
+
+The exception to this is tests, where atomics can be useful for sensing some value that you would like to assert on that is manipulated in another goroutine.
+Even here, be wary of race conditions, such as assuming that a background goroutine has executed before the test goroutine makes its assertions.
+
+## Use uber-go/atomic
+
+OK, so you've decided to use atomics.
+Use `github.com/uber-go/atomic`, rather than the built-in `sync/atomic` package.
+
+### How
+
+If your atomic is a global variable, you may declare it using its zero value:
```

Comment:
I am wondering if it would be easier to follow if we require every atomic variable to be declared as a pointer. I don't have a strong opinion here (thus my approval) but I am putting it out here just in case someone has.

Reply:
I don't really know how package-level initialization works or what the concerns are here. This works: https://go.dev/play/p/vpBoGDamE02 and reading this it seems like it would be safe https://go.dev/ref/spec#Package_initialization but I am not entirely sure